frontend/server/src/Controllers/Session.php

```diff
@@ -327,6 +329,31 @@ private static function getCurrentSessionImpl(
             if (is_null($currentUser)) {
                 throw new \OmegaUp\Exceptions\NotFoundException('userNotExist');
             }
+            $now = \OmegaUp\Time::get();
+            if (
+                !is_null(
+                    $currentUser->birth_date
+                ) && strtotime(
+                    $currentUser->birth_date
+                ) >= strtotime(
+                    '-13 year',
+                    $now
+                ) && !$currentUser->parent_verified
+            ) {
+                if (
+                    !is_null($currentUser->parent_email_verification_deadline)
+                ) {
+                                {
+                                $diff = $currentUser->parent_email_verification_deadline->time - $now;
+                                $pendingDaysForVerification = intval(
+                                    floor(
+                                        $diff / (60 * 60 * 24)
+                                    )
+                                );
+                                }
+                }
+            }
```

Comment:
`parent_email_verification_deadline` should be required here, because otherwise the banner never will be shown. It would be better throw an exception indicating that this parameter is missing
```suggestion
                if (
                    is_null($currentUser->parent_email_verification_deadline)
                ) {
                    throw new \OmegaUp\Exceptions\InvalidParameterException(
                        'parameterNotFound',
                        'parent_email_verification_deadline'
                    );     
                }
                $diff = $currentUser->parent_email_verification_deadline->time - $now;
                $pendingDaysForVerification = intval(
                    floor(
                        $diff / (60 * 60 * 24)
                    )
                );
```
.

Another validation that  should  be included is to block the account when `parent_email_verification_deadline` has expired. Maybe this is the right place for it.